### PR TITLE
[Feature] - Remote Configuration Fetch

### PR DESCRIPTION
**Pull request by DimalChandrasiri**
**Management View**
Screenshot 2020-10-23 at 13 23 41

**Status View**
Screenshot 2020-10-23 at 13 26 26

## Automation tests
 - Unit tests 
 **Test Coverage**
Screenshot 2020-10-23 at 13 30 10

### CODE REVIEW

File: apps/console/src/features/remote-repository-configuration/components/remote-fetch-details/remote-fetch-details-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+import { Grid, Icon, Modal } from "semantic-ui-react";
+import { 
+    InterfaceConfigDetails, 
+    InterfaceRemoteConfigDetails, 
+    InterfaceRemoteFetchStatus, 
+    getConfigDeploymentDetails,
+    triggerConfigDeployment
+} from "../../../remote-repository-configuration";
+
+interface RemoteFetchDetailsProps extends TestableComponentInterface {
```
Comment (brionmario) [inline apps/console/src/features/remote-repository-configuration/components/remote-fetch-details/remote-fetch-details-modal.tsx:35]: Better to change `RemoteFetchDetailsProps ` -> `RemoteFetchDetailsPropsInterface`

File: apps/console/src/features/remote-repository-configuration/components/remote-fetch-details/remote-fetch-details-modal.tsx [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+        }
+
+        setActiveIndex(newIndexes);
+    };
+
+    const getHumanizedDeployment = (date: any): string => {
+        const now = moment(new Date());
+        const receivedDate = moment(date);
+        return "Last deployed " +   moment.duration(now.diff(receivedDate)).humanize() + " ago";
+    }
```
Comment (brionmario) [inline apps/console/src/features/remote-repository-configuration/components/remote-fetch-details/remote-fetch-details-modal.tsx:88]:
```suggestion
    };
```

File: apps/console/src/features/remote-repository-configuration/components/remote-fetch-details/remote-fetch-details-modal.tsx [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+                        <Grid.Column mobile={ 8 } tablet={ 8 } computer={ 8 }>
+                            <LinkButton floated="left" onClick={ () => onClose() }>
+                                close
+                            </LinkButton>
+                        </Grid.Column>
+                        <Grid.Column mobile={ 8 } tablet={ 8 } computer={ 8 }>
+                            <PrimaryButton
+                                floated="right"
+                                data-testid={ `${ testId }-import-button` }
+                                onClick={ ()=> {
```
Comment (brionmario) [inline apps/console/src/features/remote-repository-configuration/components/remote-fetch-details/remote-fetch-details-modal.tsx:183]: Better to separate out the logic to a function. WDYT?

File: apps/console/src/features/remote-repository-configuration/components/remote-fetch-status.tsx [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+import { 
+    InterfaceRemoteConfigDetails, 
+    InterfaceRemoteRepoConfig, 
+    InterfaceRemoteRepoListResponse, 
+    getRemoteRepoConfig, 
+    getRemoteRepoConfigList, 
+    triggerConfigDeployment 
+} from "..";
+
+type RemoteFetchStatusProps = TestableComponentInterface;
```
Comment (brionmario) [inline apps/console/src/features/remote-repository-configuration/components/remote-fetch-status.tsx:37]: Rename `RemoteFetchStatusProps` -> `RemoteFetchStatusPropsInterface`

File: apps/console/src/features/remote-repository-configuration/tests/remote-fetch-noconfig.test.tsx [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+import MockEmptyAxiosResponse from "./__mocks__/mock.noconfig";
+import MockSaveConfigResponse from "./__mocks__/mock.saveConfig";
+import * as api from "../api/remote-repo-config";
+import RemoteRepoConfig from "../pages/remote-repository-config";
+
+/**
+ * This will test the remote fetch configuration component with
+ * no initial / saved configuration.
+ */
+describe("Test Suite - Remote Fetch Configuration ( Without Configuration )", () => {
```
Comment (brionmario) [inline apps/console/src/features/remote-repository-configuration/tests/remote-fetch-noconfig.test.tsx:33]: For integration test cases, we are goin with the following format.

```
ITC-<SCENARIO_NO>. <SUB_SCENARIO_NO>.<TESTCASE_NO> - [ <SCOPE> ] - <TESTCASE_DESCRIPTION>
```

Ex: `ITC-2.1.0 - [applications] - Create an Application using the wizard.`

**ITC** stands for `Integration Test Case`

Shall we go with a similar convention?

**Review by brionmario — APPROVED** [review]

File: apps/console/src/features/remote-repository-configuration/api/remote-repo-config.ts [hunk 6]
```diff
@@ -52,15 +53,15 @@ export const getRemoteRepoConfigList = (): Promise<AxiosResponse<InterfaceRemote
         });
 }
 
-export const getRemoteRepoConfig = (id: string): Promise<AxiosResponse> => {
+export const getRemoteRepoConfig = (id: string): Promise<AxiosResponse<InterfaceRemoteConfigDetails>> => {
```
Comment (brionmario) [inline apps/console/src/features/remote-repository-configuration/api/remote-repo-config.ts:56]: Shall we rename `InterfaceRemoteConfigDetails` to `RemoteConfigDetailsInterface`?

File: apps/console/src/features/remote-repository-configuration/components/remote-fetch-details/remote-fetch-details-modal.tsx (hunk 1 above)
Comment (DimalChandrasiri) [reply apps/console/src/features/remote-repository-configuration/components/remote-fetch-details/remote-fetch-details-modal.tsx]: Fixed with commit 20f0488

File: apps/console/src/features/remote-repository-configuration/components/remote-fetch-status.tsx (hunk 4 above)
Comment (DimalChandrasiri) [reply apps/console/src/features/remote-repository-configuration/components/remote-fetch-status.tsx:37]: Fixed with commit 20f0488

File: apps/console/src/features/remote-repository-configuration/api/remote-repo-config.ts (hunk 6 above)
Comment (DimalChandrasiri) [reply apps/console/src/features/remote-repository-configuration/api/remote-repo-config.ts:56]: Fixed with commit 20f0488

File: apps/console/src/features/remote-repository-configuration/tests/remote-fetch-noconfig.test.tsx (hunk 5 above)
Comment (DimalChandrasiri) [reply apps/console/src/features/remote-repository-configuration/tests/remote-fetch-noconfig.test.tsx]: Resolved with commit b6ee917

Also we might need to discuss on a convention. This task is tracked [here](https://github.com/wso2/product-is/issues/10049)

**Pull request merged**